services/supervisor/src/lib.rs

```diff
@@ -55,6 +55,41 @@
 //! The operation of starting or resuming a service is treated similarly to a configuration change
 //! and follows the same rules.
 //!
+//! ## Migrations management
+//!
+//! Supervisor service provides a functionality to perform data migrations for services.
+//! Request for migration is sent through private REST API and contains the name of instance
+//! to migrate, end artifact version to achieve after migration, and deadline height until which
```

Comment:
I think that it is a quite difficult to predict the number of blocks needed for successful migration.
In my opinion network administrators should revert broken migration by hand.

Reply:
I thought about it, but OTOH network administrators won't be able to distinguish a long-running migration from a migration that went into an infinite loop, they both will look as `Pending`.

IMHO, having the average block commitment duration, it's not that hard to calculate the approximate amount of blocks which you can afford, and then round it up. For example, if in average every block is committed in 1 second, and we don't want the migration to run for longer than 2 hours, we should set the deadline height to (current height + 7200). Taking the time between obtaining the current height and sending the request for migration, we can round it up to 7500.

So, as a conclusion, I think it's better to have such an instrument, than to not have it.

One possible improvement here may be making `deadline_height` optional (meaning that if `deadline_height` is 0, migration will never be killed due to timeout), but it'll introduce more rules to the already not very easy interaction scheme.